File: backend/app/schemas/planner.py
from pydantic import BaseModel
from app.schemas.project import ProjectMetadata


class PlannerRequest(BaseModel):
    prompt: str


class PlannerTask(BaseModel):
    id: int
    title: str


class PlannerResponse(BaseModel):
    project_name: str
    metadata: ProjectMetadata
    tasks: list[PlannerTask]

File: frontend/src/types/planner.ts
export interface ProjectMetadata {
  project_type: string;
  frameworks: string[];
  language: string;
  database: string | null;
  deployment: string | null;
  features: string[];
}

export interface PlannerTask {
  id: number;
  title: string;
}

export interface PlannerResponse {
  project_name: string;
  metadata: ProjectMetadata;
  tasks: PlannerTask[];
}